contracts/incentives/base/DistributionManager.sol

```diff
@@ -2,9 +2,9 @@
 pragma solidity 0.7.5;
 pragma experimental ABIEncoderV2;
 
-import {IAaveDistributionManager} from '../interfaces/IAaveDistributionManager.sol';
-import {SafeMath} from '../lib/SafeMath.sol';
-import {DistributionTypes} from '../lib/DistributionTypes.sol';
+import {IAaveDistributionManager} from '../../interfaces/IAaveDistributionManager.sol';
+import {SafeMath} from '../../lib/SafeMath.sol';
+import {DistributionTypes} from '../../lib/DistributionTypes.sol';
 
 /**
  * @title DistributionManager
```

Comment:
- The `AssetData` uses only `104` bits for the `index` but there is still `8` free bits before the word is filled, normally we are using `128` bits for the indexes so think we should at least use the free ones? Also, the `users` could be renamed to `userIndexes` or something like that for clarity.
- Both `getDistributionEnd()` and `DISTRIBUTION_END()` return `_distributionEnd`, are both kept for backwards compatability? 
- `_updateAssetStateInternal` fetches `emissionPerSecond` at every call, but it is not used when `block.timestamp == lastUpdateTimestamp`, could be fetched afterwards. Also the `assetConfig.lastUpdateTimestamp = uint40(block.timestamp);` can be moved out of the if as it is updated in both cases removing the else case.

Reply:
Really like your thoughts, but this PR was aimed  for making IncentivesContract abstract and would not like to touch other core parts like the Distribution Manager.

This can be done in a separate PR.

Reply:
Can you create an issue to track your optimiation?

Reply:
Added issue #8